src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -101,7 +101,7 @@ public TypeWithAnnotations GetInferredReturnType(ConversionsBase conversions, Nu
                     useSiteDiagnostics.Add(info);
                 }
             }
-            if (nullableState == null)
+            if (nullableState == null || !Binder.Compilation.RunNullableWalker)
```

Comment:
> || !Binder.Compilation.RunNullableWalker [](start = 37, length = 41)

It probably doesn't hurt to have this check, but it seems redundant. 
I think that to get here, we would need to run NullableWalker.Analyze on a method, which will in turn run NullableWalker.Analyze on lambdas inside the method.
So if the NullableWalker.Analyze is properly skipped on methods, then we shouldn't need to repeat this check.

Reply:
So this actually is hittable in DEBUG mode, because in DEBUG we always run the nullable walker, just with a temp diagnostic bag in that case. So I can't put an assertion here.

---
In reply to: [296910440](https://github.com/dotnet/roslyn/pull/36714#discussion_r296910440) [](ancestors = 296910440)